engineapi/backup_monitor.go

```diff
@@ -17,6 +17,14 @@ import (
 
 const (
 	BackupMonitorSyncPeriod = 2 * time.Second
+
+	// BackupMonitorMaxRetryPeriod is the maximum retry period when backup monitor routine
+	// encounters an error and backup stays in Pending state
+	BackupMonitorMaxRetryPeriod = 10 * time.Minute
+
+	// BackupMonitorMaxRetryCount is the maximum retry count before the backup monitor routine
+	// stops monitor and backup stays in Pending state
+	BackupMonitorMaxRetryCount = int(BackupMonitorMaxRetryPeriod / BackupMonitorSyncPeriod)
```

Comment:
I am thinking if we can use a longer retry interval rather than the backup monitor default resync period. Besides, the max retry limit is too short for large backups. I am thinking 1 or 2 hours would be better.

Reply:
Do you mean 1 to 2 hours could be the time that it takes for a large backup to be `initiated`?

Reply:
I implemented the code with exponential backoff retry mechanism by ref the [Kubernetes code](https://github.com/kubernetes/kubernetes/blob/a7d866f036a45cd4c4c8c1b183fa33c65ee1d28f/pkg/volume/csi/csi_attacher.go#L479-L511) when the Backup CR fails to `Pending` state.

The back off retry period would be 4s/8s/16s/32s/..., the maximum retry period is 2 hours.

Reply:
Let's say the max volume/backup size Longhorn supports now is 1TB. Then you can try [this cmd](https://github.com/longhorn/longhorn/issues/2507#issuecomment-857195496) with a 1TB sparse file and see how long it will take to get the fiemap. This time should be the max retry limit.